bot/exts/evergreen/emoji_count.py

```diff
@@ -17,73 +18,65 @@ class EmojiCount(commands.Cog):
+
         else:
-            msg = ''
             for key, value in emoji.items():
                 emoji_choice = random.choice(value)
                 emoji_info = f'There are **{len(value)}** emojis in the **{key}** category\n'
-                msg += f'<:{emoji_choice.name}:{emoji_choice.id}> {emoji_info}'
-            embed.description = msg
-        return embed
+                if emoji_choice.animated:
+                    msg.append(f'<a:{emoji_choice.name}:{emoji_choice.id}> {emoji_info}')
+                else:
+                    msg.append(f'<:{emoji_choice.name}:{emoji_choice.id}> {emoji_info}')
+        return embed, msg
 
     @staticmethod
-    def generate_invalid_embed(ctx: commands.Context) -> discord.Embed:
+    def generate_invalid_embed(ctx: commands.Context) -> [discord.Embed, str]:
         """Genrates error embed."""
         embed = discord.Embed(
             color=Colours.soft_red,
             title=random.choice(ERROR_REPLIES)
         )
+        msg = []
 
-        emoji_dict = {}
+        emoji_dict = defaultdict(list)
         for emoji in ctx.guild.emojis:
-            emoji_dict[emoji.name.split("_")[0]] = []
+            emoji_dict[emoji.name.split("_")[0]].append(emoji)
 
         error_comp = ', '.join(key for key in emoji_dict.keys())
-        embed.description = f"These are the valid categories\n```{error_comp}```"
-        return embed
-
-    def emoji_list(self, ctx: commands.Context, categories: dict) -> Dict:
-        """Generates an embed with the emoji names and count."""
-        out = {category: [] for category in categories}
-
-        for emoji in ctx.guild.emojis:
-            category = emoji.name.split('_')[0]
-            if category in out:
-                out[category].append(emoji)
-        return out
+        msg.append(f"These are the valid categories\n```{error_comp}```")
+        return embed, msg
 
     @commands.command(name="emoji_count", aliases=["ec"])
-    async def ec(self, ctx: commands.Context, *, emoji: str = None) -> Optional[str]:
+    async def emoji_count(self, ctx: commands.Context, *, emoji_category: str = None) -> None:
         """Returns embed with emoji category and info given by the user."""
-        emoji_dict = {}
+        emoji_dict = defaultdict(list)
 
-        for a in ctx.guild.emojis:
-            if emoji is None:
+        for emoji in ctx.guild.emojis:
+            if emoji_category is None:
                 log.trace("Emoji Category not provided by the user")
-                emoji_dict.update({a.name.split("_")[0]: []})
-            elif a.name.split("_")[0] in emoji:
+                emoji_dict[emoji.name.split("_")[0]].append(emoji)
+            elif emoji.name.split("_")[0] in emoji_category:
                 log.trace("Emoji Category provided by the user")
-                emoji_dict.update({a.name.split("_")[0]: []})
-
-        emoji_dict = self.emoji_list(ctx, emoji_dict)
+                emoji_dict[emoji.name.split("_")[0]].append(emoji)
```

Comment:
This will spam the logs based on how many emojis exist..
Just do the log first, then logic as to what you want to ignore instead of not ignore.

```py
log.trace(f"Emoji Category {'' if category_query else 'not '}provided by the user")
for emoji in ctx.guild.emojis:
    emoji_category = emoji.name.split("_")[0]

    if category_query is not None and emoji_category not in category_query:
        continue

    emoji_dict[emoji_category].append(emoji)
```
renamed emoji_category to category_query in this example, also included a suggestion to how to do the logic.